### PR TITLE
Remove mention of browser from process.runtime semantic convention

### LINKED ISSUE #2806: Remove process.runtime semconv for browser resource
**What are you trying to achieve?**

I am trying to remove the use of `process.runtime.*` attributes for browser resource. In the Client Instrumentation SIG we agreed on using only `browser.` namespace in the browser resource for identification purpose. The `process.runtime.*` predates the `browser` namespace and is added in [BrowserDetector.ts](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-resources/src/detectors/BrowserDetector.ts). Checking with few members in the SIG, it's not clear if anyone is using this detector currently.

We discussed this in the 9/14 JS SIG and [here](https://github.com/open-telemetry/opentelemetry-js/pull/3201#discussion_r961809920) - the plan is to implement a new detector for browser in a separate package and mark the current BrowserDetector in `opentelemetry-resources` package as deprecated.

So for the purpose of marking this detector as deprecated, we want to remove references to Web Browser from JavaScript Runtimes in [process.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md#javascript-runtimes) first. Note that the `process` semconv is still in Experimental state.

I will submit a PR for the removal, but creating this issue first to see if there are any concerns from the spec members.

@legendecas @martinkuba @MSNev

### PR DESCRIPTION
**Pull request by scheler**
Fixes #2806

## Changes

In the Client Instrumentation SIG we agreed on using only `browser.` namespace in the browser resource for identification purpose. In this PR, we are removing the mention of Web Browsers in the `process.runtime.*` semantic convention as they will not be used in the resource from browser instrumentation. For more context, please see https://github.com/open-telemetry/opentelemetry-specification/issues/2806

Related issues #
[#2806](https://github.com/open-telemetry/opentelemetry-specification/issues/2806)

### CODE REVIEW

File: specification/resource/semantic_conventions/process.md [hunk 1]
```diff
@@ -138,21 +138,18 @@ Examples for some Java runtimes
 
 ### JavaScript runtimes
 
-JavaScript instrumentation should fill in the values by copying from built-in runtime constants.
+JavaScript Node instrumentation should fill in the values by copying from built-in runtime constants.
```
Comment (reyang) [inline specification/resource/semantic_conventions/process.md:141]:
```suggestion
JavaScript Node.js instrumentation should fill in the values by copying from built-in runtime constants.
```

File: specification/resource/semantic_conventions/process.md [hunk 2]
```diff
@@ -138,21 +138,18 @@ Examples for some Java runtimes
...
-  - When the runtime is Node.js, fill in the constant value `nodejs`.
-  - When the runtime is Web Browser, fill in the constant value `browser`.
+  Fill in the constant value `nodejs`.
 - `process.runtime.version`:
-  - When the runtime is Node.js, fill in the value of `process.versions.node`.
-  - When the runtime is Web Browser, fill in the value of `navigator.userAgent`.
+  Fill in the value of `process.versions.node`.
 
-Examples for some JavaScript runtimes
+Example for a Node runtime:
```
Comment (reyang) [inline specification/resource/semantic_conventions/process.md:148]:
```suggestion
Example of a Node.js runtime:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,8 @@ release.
 
 - Add `browser.mobile` and `browser.language` resource attributes
   ([#2761](https://github.com/open-telemetry/opentelemetry-specification/pull/2761))
+- Remove mention of browser from `process.runtime` semantic convention
+  ([#2815](https://github.com/open-telemetry/opentelemetry-specification/pull/2815/files))
```
Comment (reyang) [inline CHANGELOG.md:23]:
```suggestion
  ([#2815](https://github.com/open-telemetry/opentelemetry-specification/pull/2815))
```

**Review by reyang — APPROVED** [review]

**Comment by scheler** [comment]
@open-telemetry/specs-approvers @open-telemetry/specs-triagers Please review.

**Comment by scheler** [comment]
not stale

File: specification/resource/semantic_conventions/process.md [hunk 4]
```diff
@@ -138,21 +138,18 @@ Examples for some Java runtimes
 
 ### JavaScript runtimes
 
-JavaScript instrumentation should fill in the values by copying from built-in runtime constants.
+JavaScript Node.js instrumentation should fill in the values by copying from built-in runtime constants.
 
 - `process.runtime.name`:
-  - When the runtime is Node.js, fill in the constant value `nodejs`.
-  - When the runtime is Web Browser, fill in the constant value `browser`.
```
Comment (arminru) [inline specification/resource/semantic_conventions/process.md:145]: I tend to agree with this comment made by @Oberon00 on the issue: https://github.com/open-telemetry/opentelemetry-specification/issues/2806#issuecomment-1256275658
We could consider leaving `process.runtime.name="browser"` in to allow for consistent handling by consumers.
Dropping the version in favor of `browser.user_agent` should be fine though IMHO.
@scheler WDYT?

Comment (carlosalberto) [reply specification/resource/semantic_conventions/process.md:145]: Ping @scheler

**Pull request closed without merging**